src/rust/engine/dep_inference/src/python/mod.rs

```diff
@@ -215,13 +255,24 @@ impl ImportCollector<'_> {
             .and_modify(|v| *v = (v.0, v.1 && self.weaken_imports))
             .or_insert(((line0 as u64) + 1, self.weaken_imports));
     }
+
+    fn handle_string_candidate(&mut self, node: tree_sitter::Node) {
+        if let Some(text) = self.extract_string(node) {
+            if !text.contains(|c: char| c.is_ascii_whitespace() || c == '\\')
+                && !self.is_pragma_ignored_recursive(node)
```

Comment:
I think this would be the place to exclude strings with quote marks or any other expressions we might want to add on the rust side. (See my other comment) Right?

Reply:
Yes, but I'd want it to be a separate PR.